app/models/nic/bmc.rb

```diff
@@ -39,6 +37,24 @@ def ensure_physical
     def enc_attributes
       @enc_attributes ||= (super + %w(username password provider))
     end
+
+    private
+
+    def bmc_proxy
+      if subnet.present?
+        proxy = subnet.proxies.find { |subnet_proxy| subnet_proxy.has_feature?('BMC') }
```

Comment:
@domcleal `subnet.proxies` is a simple array, it's not an `ActiveRecord::Relation` look [here](https://github.com/theforeman/foreman/blob/develop/app/models/subnet.rb#L196)